src/main/java/com/hubspot/jinjava/lib/tag/eager/DeferredToken.java

```diff
@@ -413,10 +418,19 @@ private static CallStack acquireMacroStack() {
       .orElse(null);
   }
 
+  private static String getFirstNonEmptyToken(List<String> strings) {
+    return Strings.isNullOrEmpty(strings.get(0)) ? strings.get(1) : strings.get(0);
```

Comment:
I think that you need to split more if you're going to take the part after a `.` because you could have `.bar.foobar` and you want `bar` not `bar.foobar`